### PR TITLE
Don't recompute var in bn decomp

### LINKED ISSUE #1988: [Inductor][CPU] Resnet50 training performance is 0.88x of eager
We observed that the performance of Resnet50 training on CPU (ICX-8380) is about 0.88x of eager. It seems to be a regression introduced by enabling the decomp of batch norm in inductor. The performance will back if we comment out [aten._native_batch_norm_legit_functional ](https://github.com/pytorch/pytorch/blob/master/torch/_inductor/decomposition.py#L70) and [aten.native_batch_norm_backward](https://github.com/pytorch/pytorch/blob/master/torch/_inductor/decomposition.py#L71) in https://github.com/pytorch/pytorch/blob/master/torch/_inductor/decomposition.py.

Reproduce command: `python -m torch.backends.xeon.run_cpu --node_id 0 --enable_jemalloc benchmarks/dynamo/torchbench.py --performance --float32 -dcpu --training --inductor  --no-skip --dashboard --only resnet50 --cold_start_latency`

### PR DESCRIPTION
**Pull request by ngimel**
Fixes https://github.com/pytorch/torchdynamo/issues/1988
Repeated `var` computation is not CSE'd for some reason.

### CODE REVIEW

**Review by Chillee — APPROVED** [review]
Can we remove that comment now?

**Comment by ngimel** [comment]
That comment is still valid, as far as I can see, the numerics of decomposition is unchanged.
